### PR TITLE
Directional Destination Tags

### PR DESCRIPTION
**Pull request by ghoshkaj**
# Issue

 #3027

## Tasklist
- [x] write Cucumber scenarios for combinations of destination, destination:ref, destination:forward, destination:backward, and oneway=yes, oneway=no, oneway=-1 (see existing scenarios)
 - [x] review
 - [x] adjust for for comments
- [x] in the profile (or maybe in profiles/lib/destinations.lua) pick up directional destination tag and set according to oneway direction (see e.g. directional hov handling)
 - [x] review
 - [x] adjust for for comments
- [x] write Cucumber scenarios for destination:ref:forward, destination:ref:backward as well
- [x] write code to address destination:ref:forward and destination:ref:backward
- [x] review
- [x] adjust for for comments

### CODE REVIEW

File: features/guidance/destination-signs-forward-backward.feature [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          | op    | OP   |     | Berlin         |                 |                     |  Hamburg             | -1     |                                      |
+          | qr    | QR   |     | Berlin;Hamburg | A1;A2           |                     |                      | no     | mis-tagged destination: not a oneway |
+
+        When I route I should get
+          | from | to | route                                                     | destinations                                    | ref   | #                         |
+          | a    | b  | AB,AB                                                     | Berlin,Berlin                                   | ,     |                           |
+          | c    | d  | CD,CD                                                     | Berlin,Berlin                                   | ,     |                           |
+          | e    | f  | EF,EF                                                     | Hamburg,Hamburg                                 | ,     |                           |
```
Comment (danpat) [inline features/guidance/destination-signs-forward-backward.feature]: This one might fail - oneway=-1 means you can only go f->e

**Comment by ghoshkaj** [comment]
Can I put the logic of forward tags and backwards tags all into the one get_destination function (https://github.com/Project-OSRM/osrm-backend/pull/3061/files#diff-b3ef18c4e3ad22970d3b6a926d7fc8bdL3)?

Right now, the logic is broken between profiles/lua.car (for backward tags) and profiles/lib/destination.lua (for forward tags) and it’s kind of ugly and unintuitive.

I’m just not sure at what level to componentize at: 

1. One function called `get_destination `that takes care of all the logic, and we just call that one function in *profiles/lua.car*
2. Four functions,`get_destination`, `get_destination_forward`,` get_destination_backward` and  `get_destination_ref`. But then we don’t really need the **profiles/lib/destination.lua** and just do all the text building within **profiles/lua.car**

It looks like nobody apart from the car profile is using this bit of text so I think it's okay if I just do Option 1.

**Comment by danpat** [comment]
@ghoshkaj Let's go with option 1 for now.  No need to overcomplicate the factoring for this, it's not like we're going to be calling `get_destination_forward` from a whole lot of places.

File: profiles/lib/destination.lua [hunk 2]
```diff
@@ -24,4 +28,48 @@ function Destination.get_destination(way)
   return rv
 end
 
+
+function Destination.get_destination_forward(way)
+  local destination_ref = way:get_value_by_key("destination:ref")
+  local destination_forward = way:get_value_by_key("destination:forward")
```
Comment (danpat) [inline profiles/lib/destination.lua]: Just to throw a wrench in the works here:

http://taginfo.openstreetmap.org/keys/destination%3Aref%3Aforward

`destination:ref` can *also* have forward/backward versions :-P  They're not super common, but there are several thousand of them out there in the wild, so we might as well add support for them while we're at it.

**Pull request merged**